presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -438,4 +439,41 @@ else if (operand >= upper) {
 
         return result;
     }
+
+    @Description("The bucket number of a value given an array of bins")
+    @ScalarFunction("width_bucket")
+    @SqlType(StandardTypes.BIGINT)
+    public static long widthBucket(@SqlType(StandardTypes.DOUBLE) double operand, @SqlType("array(double)") Block bins)
+    {
+        int numberOfBins = bins.getPositionCount();
+
+        checkCondition(numberOfBins > 0, INVALID_FUNCTION_ARGUMENT, "Bins cannot be an empty array");
+        checkCondition(!isNaN(operand), INVALID_FUNCTION_ARGUMENT, "Operand cannot be NaN");
+
+        int lower = 0;
+        int upper = numberOfBins;
+
+        int index;
+        double bin;
+
+        while (lower < upper) {
+            if (bins.getDouble(lower, 0) > bins.getDouble(upper - 1, 0)) {
+                throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Bin values are not sorted in ascending order");
+            }
+
+            index = (lower + upper) / 2;
+            bin = bins.getDouble(index, 0);
```

Comment:
Ok, thanks. I looked around for examples and saw a mixture IIRC. Will change to use `getDouble`.